src/format.ts

```diff
@@ -231,13 +230,15 @@ export class ModelFormat<T extends Entity> extends Format<T> {
 									propertyDefaultFormat = lastProperty.format;
 								}
 
+								// Ensure the target format is compiled so we can collect its paths
+								if (propertyDefaultFormat instanceof ModelFormat)
+									propertyDefaultFormat.compile();
+
 								// If the path references one or more entity properties, include paths for the property format. Otherwise, just add the path.
-								var lastIndex = formatTemplateParser.lastIndex;
```

Comment:
What's this all about?

Reply:
No idea honestly. The code between initializing `lastIndex` and restoring `formatTemplateParser.lastIndex` didn't use the regex, so it seemed like unnecessary code.